query.js

```diff
@@ -293,12 +293,35 @@ var searchMap = {
                     { travelEndMonth: { $exists: false } },
                     { travelStartMonth: { $eq: 0 } },
                     { travelEndMonth: { $eq: 0 } },
-                    {
+                    { 
+                        $and: [
+                            { travelEndYear: { $gt: startYear } },
+                            { travelStartYear: { $lt: startYear } }
+                        ],
+                    },
+                    { 
+                        $and: [
+                            { travelEndYear: { $eq: startYear } },
+                            { travelStartYear: { $eq: startYear } },
+                            { travelStartMonth: { $lte: startMonth} },
+                            { travelEndMonth: { $gte: startMonth } }
+                        ],
+                    },
+                    { 
                         $and: [
-                            { travelEndMonth: { $gte: startMonth } },
+                            { travelEndYear: { $gt: startYear } },
+                            { travelStartYear: { $eq: startYear} },
                             { travelStartMonth: { $lte: startMonth } }
-                        ]
+                        ],
+                    },
+                    {
+                        $and: [
+                            { travelStartYear: { $lt: startYear } },
```

Comment:
when travelStartYear < startYear = travelEndYear

check to see if startMonth <= travelEndMonth, then we don't care about travelStartMonth